mypy/build.py

```diff
@@ -819,8 +871,24 @@ def find() -> Optional[str]:
                     find_module_isdir_cache[pathitem, dir_chain] = isdir
                 if isdir:
                     dirs.append(dir)
-            find_module_dir_cache[dir_chain, lib_path] = dirs
-        candidate_base_dirs = find_module_dir_cache[dir_chain, lib_path]
+
+            # Third-party stub/typed packages
+            for pkg_dir in package_dirs:
+                stub_name = components[0] + '_stubs'
+                typed_file = os.path.join(pkg_dir, components[0], 'py.typed')
+                stub_typed_file = os.path.join(pkg_dir, stub_name, 'py.typed')
+                if os.path.isfile(stub_typed_file):
+                    components[0] = stub_name
+                    rest = components[:-1]
+                    path = os.path.join(pkg_dir, *rest)
```

Comment:
Should this update `dir_chain`?

Reply:
It doesn't necessarily have to, as `dir_chain` isn't used after this, unless I am overlooking something.